python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,1012 @@
+
+        """
+        return self._active
+
+    @active.setter
+    def active(self, is_active):
+        """
+        Explicitly set the active state.
+
+        There are 3 active states that can be supplied:
+
+        * ``True``: Set the item to be active
+        * ``False``: Set the item to be inactive
+        * ``None``: Clear the item's state, rely on inheritance within the tree
+        """
+        self._active = is_active
+
+    @property
+    def checked(self):
+        """
+        Boolean property to indicate that this item should be checked by
+        default when displayed in a publish UI.
+
+        .. note:: This property is shared with :py:attr:`~active` and can be
+            used interchangeably to make code more readable depending on the
+            context (with/without the UI).
+
+        Please note that the final state of the node is also affected by
+        the child tasks. Below are some examples of how this interaction
+        plays out in practice:
+
+        - If all child tasks/items return ``checked: False`` in their accept
+          method, the parent item will be unchecked, regardless
+          of the state of this property.
+
+        - If one or more child tasks return ``checked: True`` and the item
+          checked state is False, the item and all its sub-items will be
+          unchecked.
+        """
+        return self._active
+
+    @checked.setter
+    def checked(self, is_checked):
+        # setter for checked
+        self._active = is_checked
+
+    @property
+    def children(self):
+        """
+        A generator that yields the immediate :ref:`publish-api-item` children of
+        this item.
+
+        .. note:: :ref:`publish-api-item` instances are iterators so if you need
+            to traverse all descendant items, you can do this:
+
+            .. code-block:: python
+
+                for descendant in item:
+                    # process descendant item
+        """
+        for child in self._children:
+            yield child
+
+    @property
+    def context(self):
+        """
+        The :class:`sgtk.Context` associated with this item.
+
+        If no context has been explicitly set for this item, the context will be
+        inherited from the item's parent. If none of this item's parents have
+        had a context set explicitly, the publisher's launch context will be
+        returned.
+        """
+
+        if self._context:
+            return self._context
+        elif self.parent:
+            return self.parent.context
+        else:
+            return sgtk.platform.current_bundle().context
+
+    @context.setter
+    def context(self, item_context):
+        """
+        Explicitly set the context of the item.
+
+        :param item_context:
+        :return:
+        """
+        self._context = item_context
+
+    @property
+    def context_change_allowed(self):
+        """
+        ``True`` if item allows context change, False otherwise.
```

Comment:
Docstring for this property is weird and doesn't match the others. I prefer what's written for its setter, but I believe Sphinx uses the getter's docstring and ignores the setter.